File: lib/digibouquetai/site_kit.rb
module DigiBouquetAI
  module SiteKit
    BASE_URL = "https://digibouquetai.com"

    module_function

    def home_url
      BASE_URL
    end

    def page_url(slug = "")
      clean = slug.to_s.gsub(%r{\A/+|/+\z}, "")
      clean.empty? ? BASE_URL : "#{BASE_URL}/#{clean}"
    end

    def create_bouquet_url
      page_url("create-bouquet")
    end

    def ai_gift_card_url
      page_url("ai-card")
    end

    def examples_url
      page_url("bouquet-examples")
    end

    def guides_url
      page_url("guides")
    end

    def pricing_url
      page_url("pricing")
    end

    def about_url
      page_url("about")
    end
  end
end

File: nuget/DigiBouquetAISiteKit/SiteKit.cs
namespace DigiBouquetAISiteKit;

public static class SiteKit
{
    public const string BaseUrl = "https://digibouquetai.com";

    public static string HomeUrl() => BaseUrl;

    public static string PageUrl(string slug)
    {
        var clean = slug.Trim('/');
        return clean.Length == 0 ? BaseUrl : $"{BaseUrl}/{clean}";
    }

    public static string CreateBouquetUrl() => PageUrl("create-bouquet");
    public static string AiGiftCardUrl() => PageUrl("ai-card");
    public static string ExamplesUrl() => PageUrl("bouquet-examples");
    public static string GuidesUrl() => PageUrl("guides");
    public static string PricingUrl() => PageUrl("pricing");
    public static string AboutUrl() => PageUrl("about");
}
